feat(docs-infra): add stable to the list of api statuses

ajitsinghkaler opened the pull request:
Earlier api can be filtered based on security risk and deprecated now added stable as a status for better user experience

Fixes #30396

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [x] angular.io application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->
Earlier only api can be filtered based on deprecated and security
Issue Number: #30396
## What is the new behavior?
Api can be filtered using stable also

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

gkalpak commented on aio/src/app/custom-elements/api/api-list.component.ts:
```diff
@@ -117,7 +118,7 @@ export class ApiListComponent implements OnInit {
       sectionNameMatches || item.name.indexOf(query!) !== -1;
     const matchesStatus = (item: ApiItem) =>
-      status === 'all' || status === item.stability || (status === 'security-risk' && item.securityRisk);
+      status === 'all' || status === item.stability || (status === 'security-risk' && item.securityRisk) ||  (status === item.stability && !item.securityRisk);
```
The last condition seems superfluous, since the earlier `status === item.stability` would have matched already.

ajitsinghkaler replied:
I added the last one so that the security risk ones does not mix with the stable ones but you are right this condition is not right.

@gkalpak need your views on this should we add security risk items to the stable filter

ajitsinghkaler replied:
@gkalpak and tests have already been added on line 82-85 it was just that the menu item needed to be added.

If we remove security risk from the stable API then the tests need to be updated otherwise these seem enough to me please have a look at the tests and tell me if any other tests are needed

gkalpak replied:
Yes, security risk is orthogonal to stability, so `stable` should include `securityRisk` items as well. I.e. the condition should remain: `status === 'all' || status === item.stability || (status === 'security-risk' && item.securityRisk)`

Fair enough about the tests. Ideally we should have e2e tests covering this feature, but we don't 😅.
(If you want to add some e2e tests, I won't stop you, but it isn't required for this PR :grin:)

gkalpak reviewed (changes requested):
Thx, @ajitsinghkaler! Can you also add some tests for the new feature?

mary-poppins commented:
You can preview f013be0 at https://pr34981-f013be0.ngbuilds.io/.

ajitsinghkaler committed:
feat(docs-infra): add stable to the list of api statuses

Earlier api can be filtered based on security risk and deprecated now added stable as a status for better user experience

Fixes #30396

ajitsinghkaler commented:
@gkalpak made the changes I'll add e2e tests later will make a new issue for that

mary-poppins commented:
You can preview 722132c at https://pr34981-722132c.ngbuilds.io/.
